masonry/src/passes/update.rs

```diff
@@ -416,17 +416,14 @@ fn update_new_widgets(
     }
     state.item.children_changed = false;
 
-    // This will recursively call WidgetPod::lifecycle for all children of this widget,
-    // which will add the new widgets to the arena.
     {
-        let mut ctx = LifeCycleCtx {
-            global_state,
-            widget_state: state.item,
+        let mut ctx = RegisterCtx {
             widget_state_children: state.children.reborrow_mut(),
             widget_children: widget.children.reborrow_mut(),
         };
-        let event = LifeCycle::Internal(crate::InternalLifeCycle::RouteWidgetAdded);
-        widget.item.lifecycle(&mut ctx, &event);
+        // The widget will call `RegisterCtx::register_child` on all its children,
+        // which will add the new widgets to the arena.
+        widget.item.register_children(&mut ctx);
     }
 
     if state.item.is_new {
```

Comment:
Just scanning here, does this mean that parents get their `WidgetAdded` before their children?

I'm not sure which way around is best here...

Reply:
It does. We could switch it up by changing the order of paragraphs in that function. I'm adding this to the list of things I need to document.